spreadsheet/formula/functions_test.go

```diff
@@ -0,0 +1,328 @@
+	d2Style.SetNumberFormat("MMM DD")
+	sheet.Cell("A15").SetStyle(d2Style)
+
+	d3Style := ss.StyleSheet.AddCellStyle()
+	d3Style.SetNumberFormat("MM/YY")
+	sheet.Cell("A16").SetStyle(d3Style)
+
+	d4Style := ss.StyleSheet.AddCellStyle()
+	d4Style.SetNumberFormat("MM/DD/YY\\ HH:MM\\ AM/PM")
+	sheet.Cell("A17").SetStyle(d4Style)
+
+	d4Style = ss.StyleSheet.AddCellStyle()
+	d4Style.SetNumberFormat("MM/DD/YYYY\\ HH:MM:SS")
+	sheet.Cell("A18").SetStyle(d4Style)
+
+	d5Style := ss.StyleSheet.AddCellStyle()
+	d5Style.SetNumberFormat("MM\\-DD")
+	sheet.Cell("A19").SetStyle(d5Style)
+
+	d6Style := ss.StyleSheet.AddCellStyle()
+	d6Style.SetNumberFormat("HH:MM:SS\\ AM/PM")
+	sheet.Cell("A20").SetStyle(d6Style)
+
+	d7Style := ss.StyleSheet.AddCellStyle()
+	d7Style.SetNumberFormat("HH:MM\\ AM/PM")
+	sheet.Cell("A21").SetStyle(d7Style)
+
+	d8Style := ss.StyleSheet.AddCellStyle()
+	d8Style.SetNumberFormat("HH:MM:SS")
+	sheet.Cell("A22").SetStyle(d8Style)
+
+	d9Style := ss.StyleSheet.AddCellStyle()
+	d9Style.SetNumberFormat("HH:MM")
+	sheet.Cell("A23").SetStyle(d9Style)
+
+	sciStyle := ss.StyleSheet.AddCellStyle()
+	sciStyle.SetNumberFormat("##0.00E+00")
+	sheet.Cell("A24").SetStyle(sciStyle)
+
+	incorrectStyle := ss.StyleSheet.AddCellStyle()
+	incorrectStyle.SetNumberFormat("incorrect style")
+	sheet.Cell("A25").SetStyle(incorrectStyle)
+
+// for testing alignments ("prefix" function)
+	leftStyle := ss.StyleSheet.AddCellStyle()
+	leftStyle.SetHorizontalAlignment(sml.ST_HorizontalAlignmentLeft)
+	sheet.Cell("B2").SetStyle(leftStyle)
+
+	rightStyle := ss.StyleSheet.AddCellStyle()
+	rightStyle.SetHorizontalAlignment(sml.ST_HorizontalAlignmentRight)
+	sheet.Cell("B3").SetStyle(rightStyle)
+
+	centerStyle := ss.StyleSheet.AddCellStyle()
+	centerStyle.SetHorizontalAlignment(sml.ST_HorizontalAlignmentCenter)
+	sheet.Cell("B4").SetStyle(centerStyle)
+
+	fillStyle := ss.StyleSheet.AddCellStyle()
+	fillStyle.SetHorizontalAlignment(sml.ST_HorizontalAlignmentFill)
+	sheet.Cell("B5").SetStyle(fillStyle)
+
+	ctx := sheet.FormulaContext()
+
+// for testing protected cells
+	ctx.SetLocked("A1", true)
+	ctx.SetLocked("B1", false)
+
+// for testing widths
+	sheet.Column(1).SetWidth(1.5 * measurement.Inch)
+	sheet.Column(2).SetWidth(2.5 * measurement.Inch)
+
+	runTests(t, ctx, td)
+}
+
+func TestChoose(t *testing.T) {
+	td := []testStruct{
+		{`=CHOOSE(A1,B1,B2,B3)`, `value1 ResultTypeString`},
+		{`=CHOOSE(A2,B1,B2,B3)`, `value2 ResultTypeString`},
+		{`=CHOOSE(A3,B1,B2,B3)`, `value3 ResultTypeString`},
+		{`=CHOOSE(A3,B1,B2)`, `#VALUE! ResultTypeError`},
+	}
+
+	ss := spreadsheet.New()
+	sheet := ss.AddSheet()
+
+	sheet.Cell("A1").SetNumber(1)
+	sheet.Cell("A2").SetNumber(2)
+	sheet.Cell("A3").SetNumber(3)
+
+	sheet.Cell("B1").SetString("value1")
+	sheet.Cell("B2").SetString("value2")
+	sheet.Cell("B3").SetString("value3")
+
+	ctx := sheet.FormulaContext()
+
+	runTests(t, ctx, td)
+}
+
+func TestColumn(t *testing.T) {
+	td := []testStruct{
+		{`=COLUMN(A1)`, `1 ResultTypeNumber`},
```

Comment:
Are these default results? Seems like the test is run on a freshly created empty file

Reply:
yes, column A is always 1, column B is always 2, no need to modify freshly created empty file